hosted/helpers/helper_common.go

```diff
@@ -129,6 +128,15 @@ func WaitUntilClusterIsReady(cluster *management.Cluster, client *rancher.Client
 	return client.Management.Cluster.ByID(cluster.ID)
 }
 
+// WaitClusterToBeUpgraded waits for the cluster to be upgraded and then returns the updated cluster
+//
+//	TODO(pvala): To be used in GKE and EKS and replace [GKE/EKS]Config with [GKE/EKS]Status.UpstreamSpec for testing the updates.
+func WaitClusterToBeUpgraded(client *rancher.Client, clusterID string) (*management.Cluster, error) {
```

Comment:
Not sure of this new function, if existing one can be used

Reply:
The existing one is good, but then after every call, we would need to fetch the cluster again to do the testing since we are moving to using `*Status.UpstreamSpec`, it seemed fitting to have a separate function that waited and returned newly created cluster.

Reply:
> to have a separate function that waited and returned an updated cluster.

Is it tested, I believe _cluster_ object would always give current cluster state

Reply:
It is tested. `cluster` object needs to be reloaded to fetch the latest *Status.UpstreamSpec object. But your comment was a good catch since it helped me realize that this function is not useful in this context, we can do without it, so I have removed it.